modules/apps/wiki/wiki-web/src/main/java/com/liferay/wiki/web/internal/display/context/WikiListPagesDisplayContext.java

```diff
@@ -535,14 +547,37 @@ else if (navigation.equals("draft-pages") ||
 
 			int wikiPageStatus = status;
 
-			searchContainer.setResultsAndTotal(
-				() -> WikiPageServiceUtil.getPages(
-					themeDisplay.getScopeGroupId(), wikiPageDraftUserId,
-					_wikiNode.getNodeId(), wikiPageStatus,
-					searchContainer.getStart(), searchContainer.getEnd()),
-				WikiPageServiceUtil.getPagesCount(
-					themeDisplay.getScopeGroupId(), wikiPageDraftUserId,
-					_wikiNode.getNodeId(), wikiPageStatus));
+			if (wikiPageDraftUserId == 0) {
+				searchContainer.setResultsAndTotal(
+					() -> WikiPageServiceUtil.getPages(
+						themeDisplay.getScopeGroupId(), wikiPageDraftUserId,
+						_wikiNode.getNodeId(), wikiPageStatus,
+						searchContainer.getStart(), searchContainer.getEnd()),
+					WikiPageServiceUtil.getPagesCount(
+						themeDisplay.getScopeGroupId(), wikiPageDraftUserId,
+						_wikiNode.getNodeId(), wikiPageStatus));
+			}
+			else {
+				searchContainer.setResultsAndTotal(
+					(List<WikiPage>)WikiPageLocalServiceUtil.dslQuery(
+						DSLQueryFactoryUtil.select(
+							WikiPageTable.INSTANCE
+						).from(
+							WikiPageTable.INSTANCE
+						).where(
+							WikiPageTable.INSTANCE.groupId.eq(
+								themeDisplay.getScopeGroupId()
+							).and(
+								WikiPageTable.INSTANCE.statusByUserId.eq(
+									wikiPageDraftUserId)
+							).and(
+								WikiPageTable.INSTANCE.nodeId.eq(
+									_wikiNode.getNodeId())
+							).and(
+								WikiPageTable.INSTANCE.status.eq(wikiPageStatus)
+							)
+						)));
```

Comment:
Sorry for not being explicit about this in the previous review: can you move this logic to a new method in WikiPageLocalService?

Reply:
Due to how JSONObject.toString handles type double as a whole number, the ".0" from the value is truncated before being stored in the SocialActivity.extraData column.  We may end us showing unapproved attachments, as they are stored as version "1" and not "1.0".  This only applies to whole numbers afaik, so I added this small conditional.